src/protocol/codecs/topic.rs

```diff
@@ -81,6 +107,24 @@ impl TopicCodec {
 
         Ok(topics)
     }
+
+    /// Marshall topics to a byte stream.
+    fn marshall_topics(&mut self, topics: Vec<Topic>) -> BytesMut {
+        let mut raw_data = BytesMut::new();
+        let num_topics = topics.len() as u8;
```

Comment:
I'm taking there is no chance for this to blow up? Maybe there should be some asserts here?

Reply:
raw_data can panic technically while doing these `put` operations, but we cannot control that (return Err if that happens),

but other than that, I don't see any need for error handling here. A vector of topics will always be a valid vector of topics, no?

Reply:
What I meant was this conversion `len() as u8`. Can len() be longer than 255?

Reply:
> Can len() be longer than 255?

Today, no. I think the largest value for the `len()` could be 5. I could leave a comment, so no one worries in the future

Reply:
Good catch!
So, [here](https://github.com/algorand/go-algorand/blob/8a335bb2b10ea3cb6e9471526701bcaa4089b53d/network/topics.go#L43):

```
// Topics is an array of type Topic
// The maximum number of topics allowed is 32
// Each topic key can be 64 characters long and cannot be size 0
type Topics []Topic
```
u8 is always okay for `num_topics` and `key_len`.

But this encoding/decoding won't work for the data part if the data size is longer than sizeof(u8).
So far, we are safe for this MsgOfInterest message.

I'll leave a TODO before lines 94 and 118 to check whether the variable length encoder/decoder is needed for future mesages.